src/commands/deployments/connectToGitHub.ts

```diff
@@ -0,0 +1,25 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { DeploymentsTreeItem, editScmType } from "vscode-azureappservice";
+import { AzureTreeItem, IActionContext } from "vscode-azureextensionui";
+import { ScmType } from "../../constants";
+import { SiteTreeItem } from "../../explorer/SiteTreeItem";
+import { WebAppTreeItem } from "../../explorer/WebAppTreeItem";
+import { ext } from "../../extensionVariables";
+
+export async function connectToGitHub(this: IActionContext, node?: SiteTreeItem | DeploymentsTreeItem): Promise<void> {
+    if (!node) {
+        node = <SiteTreeItem>await ext.tree.showTreeItemPicker(WebAppTreeItem.contextValue);
+    }
+    await editScmType(node.root.client, node, this, ScmType.GitHub);
+    if (node instanceof SiteTreeItem) {
+        const children: AzureTreeItem[] = await node.getCachedChildren();
```

Comment:
`deploymentsNode` is a public property on WebAppTreeItem - no need to search the children. (You don't support slots yet, so the type of node can be `WebAppTreeItem` instead of `SiteTreeItem`. Please file an issue to support slots separately so that we can get this merged ASAP)